src/postgres/tree/PostgresDatabaseTreeItem.ts

```diff
@@ -98,11 +104,7 @@ export class PostgresDatabaseTreeItem extends AzureParentTreeItem<ISubscriptionC
 
         if (!username || !password) {
             if (warnBeforePrompting) {
-                await ext.ui.showWarningMessage(
-                    localize('mustEnterUsernameAndPassword', 'You must enter the username and password for server "{0}" to continue.', this.parent.label),
-                    { modal: true },
-                    { title: localize('continue', 'Continue') }
-                );
+                throw new PromptForCredentialsError();
```

Comment:
You're not using most of this function for this case. I suggest you use `getCredentialsFromKeytar` directly inside `loadMoreChildrenImpl`.

At that point, `getCredentials` is only called once from `getPostgresCredentials.ts` and it always prompts, thus I feel like the name would make more sense if it were something like `promptForCredentials` or `enterCredentials`